app/javascript/listings/components/CategoryLinks.jsx

```diff
@@ -0,0 +1,29 @@
+import { h } from 'preact';
+import PropTypes from 'prop-types';
+
+const CategoryLinks = ({ categories, onClick, selectedCategory }) => {
+  return (
+    <section>
+      {categories.map((category) => (
+        <a
+          href={`/listings/${category.slug}`}
+          id={`category-link-${category.id}`}
+          className={category.slug === selectedCategory ? 'selected' : ''}
+          onClick={onClick}
```

Comment:
This fixes the issue where a category wasn't being selected.

```suggestion
          onClick={(e) => {
            onClick(e, category.slug);
          }}
```

Reply:
✅